### ISSUE TITLE
Fail to Evaluate More Than Two Disjunction Predicates

### ISSUE THREAD

**Issue by apavlo** [cmu-db/terrier#1058]
The DBMS fails to properly evaluate `WHERE` clauses with more than two `OR` predicates. It appears that only the first two predicates are considered and then rest are ignored:

```
terrier=# CREATE TABLE xxx (id INT);
CREATE TABLE
terrier=# INSERT INTO xxx VALUES (1),(2),(3),(4);
INSERT 0 4
terrier=# SELECT * FROM xxx WHERE id=1 OR id=2 OR id=3 OR id=4;
 id 
----
  1
  2
(2 rows)
```

The first step is to figure out where the expression tree includes all of the predicates at the different stages of the execution path (e.g., parser, binder, optimizer, codegen).

**Comment by thepinetree** [comment]
Can we close this issue?

**Comment by wuwenw** [comment]
Fixed in #1064
